app/controllers/flow/partner_handler.rb

```diff
@@ -1,13 +1,33 @@
 module Flow
-  class PartnerHandler
-    class << self
-      def model(session_data)
-        PartnerForm.new session_data.slice(*PartnerForm::ATTRIBUTES.map(&:to_s))
-      end
+  class PartnerHandler < GenericHandler
+    def model_from_session(session_data)
+      session_keys = @form_class::ATTRIBUTES.map(&:to_s).map { "partner_#{_1}" }
+      session_attributes = session_data.slice(*session_keys)
+      transformed_session_attributes = session_attributes.transform_keys { _1.gsub("partner_", "") }
+      partner_form_class.new(transformed_session_attributes).tap { modify(_1, session_data) }
+    end
+
+    def model_from_params(params, session_data)
+      form_params = params.fetch(partner_form_class_name.underscore, {})
+      partner_form_class.new(form_params.permit(*@form_class::ATTRIBUTES)).tap { modify(_1, session_data) }
+    end
 
-      def form(params, _session_data)
-        PartnerForm.new(params.fetch(:partner_form, {}).permit(*PartnerForm::ATTRIBUTES))
+    def extract_attributes(form)
+      form.attributes.transform_keys { "partner_#{_1}" }
+    end
+
+    # Ensure that the model's class name starts with 'Partner' so that it gets its own
+    # key for validation error i18n.
+    def partner_form_class
+      if Object.const_defined?(partner_form_class_name)
+        partner_form_class_name.constantize
+      else
+        Object.const_set(partner_form_class_name, Class.new(@form_class))
       end
```

Comment:
frankly I don't understand this method. Why do we need it? It looks very low-level. I would have expected partner_form_class_name.constantize to always work...?